.github/workflows/coverage.yml

```diff
@@ -23,10 +23,21 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
 
+      # NPM started understanding yarn.lock file starting from v7
+      - name: Update NPM
+        shell: bash
+        run: |
+          export NPM_VERSION=$(npm -v)
+          export IS_NPM_SUITABLE=$(node -p "parseInt(process.env.NPM_ROOT, 10) >= 7")
+          if [ "$IS_NPM_SUITABLE" == "false" ]; then
+            echo "NPM needs upgrading!"
+            npm i npm@7 -g
+          fi
```

Comment:
If we know that 10x 12x and 14x will sure need it and others don't, we can probably do
``` 
if: contains(["10.x", "12.x", "14.x"], matrix.node-version
run: npm i npm@7 -g
```

Reply:
While it is likely that we will continue to get v6 for those versions, we cannot be sure. Detection takes only milliseconds and upgrading happens in under 5 seconds on Linux and under 20 seconds on Windows.